lib/tapioca/gem/listeners/source_location.rb

```diff
@@ -41,25 +55,31 @@ def on_method(event)
         def add_source_location_comment(node, file, line)
           return unless file && line
 
-          gem = @pipeline.gem
           path = Pathname.new(file)
           return unless File.exist?(path)
 
           # On native extensions, the source location may point to a shared object (.so, .bundle) file, which we cannot
           # use for jump to definition. Only add source comments on Ruby files
           return unless path.extname == ".rb"
 
-          path = if path.realpath.to_s.start_with?(gem.full_gem_path)
-            "#{gem.name}-#{gem.version}/#{path.realpath.relative_path_from(gem.full_gem_path)}"
+          gem = @pipeline.gem
+          realpath = path.realpath.to_s
+
+          # If the definition source location is not from the current gem, look for it in dependencies
+          gem = @gem_specs[realpath] unless realpath.start_with?(gem.full_gem_path)
+          return if gem.nil?
+
+          path = if gem.default_gem?
+            "#{gem.name}//#{path.realpath.relative_path_from(RbConfig::CONFIG["rubylibdir"])}"
           else
-            path.sub("#{Bundler.bundle_path}/gems/", "").to_s
+            "#{gem.name}//#{path.realpath.relative_path_from(gem.full_gem_path)}"
           end
```

Comment:
I think it would be best to encapsulate the logic of finding the relative path of a given file path in the `GemSpec` class, so that we don't have to know about libdirs and default gems, etc here.

Something like:
```
path = gem.relative_path_for(file)
```